pkg/controller/perconaservermongodb/status.go

```diff
@@ -178,9 +178,9 @@ func (r *ReconcilePerconaServerMongoDB) updateStatus(cr *api.PerconaServerMongoD
 	cr.Status.Host = host
 
 	switch {
-	case replsetsStopping > 0 || cr.Status.Mongos.Status == api.AppStateStopping || cr.ObjectMeta.DeletionTimestamp != nil:
+	case replsetsStopping > 0 || (cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status == api.AppStateStopping) || cr.ObjectMeta.DeletionTimestamp != nil:
 		cr.Status.State = api.AppStateStopping
-	case replsetsPaused == len(repls) && cr.Status.Mongos.Status == api.AppStatePaused:
+	case replsetsPaused == len(repls) && (cr.Spec.Sharding.Enabled && cr.Status.Mongos.Status == api.AppStatePaused):
```

Comment:
if `cr.Spec.Sharding.Enabled == false` we will never get Paused state

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/713/commits/205f268746056d69e107f12d322f6a3a7e648d16